tests-arquillian/src/test/java/org/jboss/weld/tests/extensions/SimpleExtension.java

```diff
@@ -34,7 +34,7 @@ public static boolean isObservedBeforeBeanDiscovery() {
         return observedBeforeBeanDiscovery;
     }
 
-    public void observeBeforeShutdown(@Observes BeforeShutdown beforeShutdown, BeanManager beanManager) {
+    public void observeBeforeShutdown(@Observes BeforeShutdown beforeShutdown, ELAwareBeanManager beanManager) {
```

Comment:
ATM, Weld is checking strictly for `BeanManager` type as an optional argument in observer methods (not even `WeldManager` works).
I am not sure why that is the case but it should be fairly simple to fix.
I have created https://issues.redhat.com/browse/WELD-2811 to track this and will fix it for 6.x as well as 7.x

Reply:
I would guess that at least specification defined types should work, so not just `BeanManager`, but also `BeanContainer` and `ELAwareBeanManager`.

Reply:
I agree. I've already sent PRs for this, see https://github.com/weld/core/pull/3134